internal/app/machined/pkg/system/services/apid.go

```diff
@@ -153,11 +150,12 @@ func (o *APID) Runner(r runtime.Runtime) (runner.Runner, error) {
 		runner.WithStdin(stdin),
 		runner.WithLoggingManager(r.Logging()),
 		runner.WithContainerdAddress(constants.SystemContainerdAddress),
-		runner.WithContainerImage(image),
 		runner.WithEnv(env),
 		runner.WithOCISpecOpts(
 			oci.WithHostNamespace(specs.NetworkNamespace),
 			oci.WithMounts(mounts),
+			oci.WithRootFSPath("/opt/apid"),
```

Comment:
Is this really required? `WithRootFSReadonly` should make the standard root path of the container read only.

Reply:
we need to have binary available in the rootfs itself? it can't run from empty directory

Reply:
IIUC `containerd` should create an empty dir for it somewhere in `/system`, then we just bind mount the binary into it. Kind of like a `docker run -v /bin/binary:/bin/binary --entrypoint=/bin/binary alpine`.

Reply:
we figured out this in a conversation, tl;dr is that we don't have anything in `/system` for this contianer

in the future we might even drop system containerd and use `runc` directly